### PR TITLE
Fix model field defaults to an empty string and is determined to be null

### PR DESCRIPTION
**Pull request by zGaron**
as comment.
if the field is  " not null default '' " in database.
model.zep:2091 while return null.

Thanks

### CODE REVIEW

**Comment by sergeyklay** [comment]
@zGaron Could you please provide a small script to reproduce the issue?

**Comment by zGaron** [comment]
@sergeyklay 

just in update

```php
        $db = Di::getDefault()->get('db');
        
        $table_name = "bugfix_12507";
        
        
        $ddl = "CREATE TABLE {$table_name}(id INT PRIMARY KEY NOT NULL AUTO_INCREMENT, test_field VARCHAR(12) DEFAULT '' NOT NULL);";
        $ret = $db->execute($ddl);
        if (!$ret) {
            var_dump('create test table failed!');
        }
        
        
        $modelClass = "Bugfix12507";
        $modelCde   = "Class {$modelClass} extends Phalcon\\Mvc\\Model {" . PHP_EOL .
            "public \$id;" . PHP_EOL .
            "public \$test_field;" . PHP_EOL .
            "public function getSource(){ return '{$table_name}';}" . PHP_EOL .
            "}" . PHP_EOL;
        eval($modelCde);
    
        /**
         * @var Model $model
         */
        $model = new $modelClass();
        $model->test_field = '';
        $ret = $model->create();
        if(!$ret) {
            echo "on create!", "<br/>";
            $message = $model->getMessages();
            /**
             * @var $msg Model\Message
             */
            foreach ($message as $msg) {
                echo $msg->getMessage(), '<br/>';
            }
        }
    
        $model->test_field = time();
        $model->save();
    
        $model = $modelClass::findFirst(1);
        $model->test_field = '';
        $ret = $model->save();
        
        if(!$ret) {
            echo "on update!", "<br/>";
            $message = $model->getMessages();
            /**
             * @var $msg Model\Message
             */
            foreach ($message as $msg) {
                echo $msg->getMessage(), '<br/>';
            }
        }
        
        $db->execute('DROP TABLE '.$table_name);
```

**Comment by zGaron** [comment]
hi @sergeyklay 
I  have added a unit test for this PR, please see if this is the one you want.

**Comment by sergeyklay** [comment]
@zGaron Could you please submit this PR to the 3.0.x branch. I'll merge 3.0.x with 3.1.x later. Also update please CHANGELOG.md

**Comment by zGaron** [comment]
@sergeyklay  ok

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 - Fixed `Phalcon\Cache\Backend\Libmemcached::queryKeys` to correct query the existing cached keys [#11024](https://github.com/phalcon/cphalcon/issues/11024)
 - Fixed building extension for ImageMagick 7 [mkoppanen/imagick#180](https://github.com/mkoppanen/imagick/issues/180)
 - Fixed `Phalcon\Cache\Backend\Redis::save` to allow save data termlessly [#12327](https://github.com/phalcon/cphalcon/issues/12327)
+- Fixed default value of the model is not correct properly when updating. [#12507](https://github.com/phalcon/cphalcon/issues/12507)
```
Comment (sergeyklay) [inline CHANGELOG.md]: The 3.0.3 already released. Use 3.0.4 please

**Review by sergeyklay — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (zGaron) [inline CHANGELOG.md]: XD
got it

**Comment by zGaron** [comment]
@sergeyklay 
Can I use [ci skip] when I just modify the CHANGELOG.md ?

**Review by sergeyklay — APPROVED** [review]

**Comment by sergeyklay** [comment]
Thanks

**Pull request merged**